File: BoardGames/HitOrMiss/include/libpython/board_view.py
def build_separator_line( board_size ) :
  line = None

  if board_size > 0 :
    line = '   '

    line += '+---' * board_size
    line += '+'

  return line

def build_header_line( board_size ) :
  line = None

  if board_size > 0 :
    line   = '   '
    letter = 0
    for _ in range( board_size ) :
      line += '  {0} '.format( chr( letter + ord( 'A' ) ) )
      letter += 1
    line += ' '

  return line

def build_row_line( board, row_number ) :
  line = None

  if row_number in range( 1, len( board ) + 1 ) :
    line = ' {0} '.format( row_number )
    row  = board[ row_number - 1 ]

    for column_index in range( len( row ) ) :
      piece = row[ column_index ]
      piece = piece if ( piece != 'x' and piece != '' ) else '.'

      line += '| {0} '.format( piece )
    line += '|'

  return line

def build_board_lines( board ) :
  lines = None

  if type( board ) is list :
    board_size = len( board )
    if board_size > 0 :
      lines     = list()
      first_row = board[ 0 ]
      row_size  = len( first_row )

      lines.append( build_header_line    ( row_size ) )
      lines.append( build_separator_line ( row_size ) )

      for row_number in range( 1, board_size + 1 ) :
        lines.append( build_row_line       ( board, row_number ) )
        lines.append( build_separator_line ( row_size          ) )

  return lines

File: BoardGames/HitOrMiss/include/libruby/board_view.rb
# 2016.03.25: DRY - Use blocks to build line, where block is separator, header or row.

def build_separator_line( board_size )
  # ( not board_size.nil? and board_size > 0 ) ? '   +' + ( '---+' * board_size ) : nil
  line = nil

  if not board_size.nil? and board_size > 0
    line = '   '

    board_size.times do
      line += "+---"
    end
    line += '+'

  end

  return line
end

def build_header_line( board_size )
  line = nil

  if not board_size.nil? and board_size > 0
    line = '   '

    board_size.times do | iteration |
      letter = ( iteration + 'A'.ord ).chr
      line += "  #{ letter } "
    end
    line += ' '
  end

  return line
end

def build_row_line( board, row_number )
  line = nil

  if not board.nil? and ( 1 .. board.size ).include?( row_number )
    line = " #{ row_number } "
    row  = board[ row_number - 1 ]

    row.each do | cell |
      piece = ( cell == 'x' or cell == '' ) ? '.' : cell
      line += "| #{ piece } "
    end
    line += '|'
  end

  return line
end

def build_board_lines( board )
  lines = nil

  if board.instance_of?( Array ) and board.size > 0
    first_row = board.first
    row_size  = first_row.size
    lines     = Array.new

    separator_line = build_separator_line ( row_size )
    header_line    = build_header_line    ( row_size )
    lines << header_line << separator_line

    board.each_index do | row_index |
      row_number = row_index + 1
      row_line   = build_row_line( board, row_number )
      lines << row_line << separator_line
    end
  end

  return lines
end
